### ISSUE TITLE
DragImagePreview does not work with touch backend

### ISSUE THREAD

**Issue by pandaiolo** [react-dnd/react-dnd#2206]
**Describe the bug**

When using the touch backend, there is no preview of the dragged element. Using DragPreviewImage does not seem to work.

**Reproduction**

You can find a codesandbox  here: https://codesandbox.io/s/festive-borg-z01w6

This is just the DragPreviewImage example, with touch backend instead of html5

1. Go to https://codesandbox.io/s/festive-borg-z01w6
2. Drag with image
3. Div opacity is decreased, showing that dragging has started, but no preview image is visible

**Expected behavior**

Drag image preview should be visible like with html5 backend.

**Desktop (please complete the following information):**

- OS: iOS 13.4
- Browser Safari

- OS: Mac 10.15.4
- Browser: Chrome with DevTools Touch simulator
- Version: 80.0.3987.149 (Official Build) (64-bit)

**Comment by pandaiolo** [comment]
I see someone opened  a SO question here as well: https://stackoverflow.com/questions/60960230/cant-see-the-item-that-i-am-dragging-using-react-dnd

**Comment by jk171505** [comment]
I'm having the same problem, any updates on this?

**Comment by wunnle** [comment]
Same problem here.

**Comment by Nickfis** [comment]
Same problem for me as well.

**Comment by kpettinga** [comment]
Same problem only with the MouseBackend from https://github.com/zyzo/react-dnd-mouse-backend

**Comment by eeeman1** [comment]
same problem when solved?

**Comment by jonathonbalogh** [comment]
+1 - same issue.

**Comment by jpodpro** [comment]
same!

**Comment by farhantahir** [comment]
Facing this issue any solution ?

**Comment by smooth-lee** [comment]
I solved it using react-dnd-preview.

https://louisbrunner.github.io/dnd-multi-backend/packages/react-dnd-preview/

**Comment by mathieuedet** [comment]
Same issue. Is it a bug of DragImagePreview ? @smooth-lee 's solution is the only alternative ?

**Comment by ddykhoff** [comment]
I believe the root cause here is that the HTML5 backend, which utilizes the HTML5 Drag and Drop API, offloads the drag preview generation to the browser API. Since in this case the TouchBackend is being used, this feature doesn't exist. Thus no drag preview is automatically generated for you.

I believe that the `react-dnd` framework solution would be to use `useDragLayer` and make your own CustomDragLayer component. There is an example [here](https://react-dnd.github.io/react-dnd/examples/drag-around/custom-drag-layer) on how to do so. Although, this is essentially what the `react-dnd-preview` library handles for you as you can see [here](https://github.com/LouisBrunner/dnd-multi-backend/blob/master/packages/react-dnd-preview/src/usePreview.js#L56).

Personally, I opted to implement the custom drag layer myself, as I already had for another DND use case in my app and also to minimize pulling in additional dependencies.

**Comment by rpivo** [comment]
After trying @ddykhoff's suggestion to use `useDragLayer` to create a preview for touch events, I was also able to make this work without installing any external dependencies.

**Comment by tommasini** [comment]
I tried the solution with `useDragLayer` and the react-dnd-preview, both of them shows me that I could drag with the touch backend, but I can't drop neither the hover that I set on useDrop works, any one can share a solution or help me?

Thank you!

**Comment by dooleyb1** [comment]
Still experiencing this issue, when dragging on mobile I am not able to see the element being dragged. Anyone find a decent solution?

**Comment by MikeDrewitt** [comment]
Still broken. You can use the `useDragLayer` to show a preview that way. But I find it less performant in general, and annoying that you need to implement it at all when all you want is an image.

**Comment by kirlen** [comment]
@tommasini I'm having the same problem (drop/hover not working with touch backend). Did you ever get anywhere with this?

**Comment by tommasini** [comment]
No, I believe I went to other library at that time 😔 good luck 🍀

**Comment by Roman1510** [comment]
Still no resolution except using **custom drag layer** with `useDragLayer`. (tested on Chrome MacOS)
Chrome on Windows works absolutely fine with the preview that's automatically rendered based on `ref`
